servers/curated-corpus-api/src/admin/resolvers/queries/UrlMetadata/index.ts

```diff
@@ -0,0 +1,52 @@
+import { UserInputError } from '@pocket-tools/apollo-utils';
+
+import { UrlMetadata } from 'content-common';
+
+import config from '../../../../config';
+import { IAdminContext } from '../../../context';
+import {
+  convertParserJsonToUrlMetadata,
+  derivePublisher,
+  fetchUrlMetadata,
+} from './lib';
+
+/**
+ * main entry point for the resolver for the getUrlMetadata query. validates
+ * the url provided.
+ *
+ * @param parent
+ * @param param1
+ * @param ctx
+ * @returns UrlMetadata object
+ */
+export const getUrlMetadata = async (
+  parent,
+  { url },
+  context: IAdminContext,
+): Promise<UrlMetadata> => {
+  try {
+    // validate url by throwing if url format is incorrect
+    new URL(url).toString();
+  } catch (error) {
+    throw new UserInputError(`${url} is not a valid url`);
+  }
+
+  // attempts to retrieve metadata about the given url from an external
+  // metadata scraper service. returns an empty object if the external
+  // service errors.
+  const metadataJson = await fetchUrlMetadata(
+    url,
+    config.metadataParser.endpoint,
+    config.metadataParser.apiKey,
+  );
+
+  // returns a UrlMetadata formatted object, at minimum containing a URL and
+  // a domain.
+  const urlMetadata = convertParserJsonToUrlMetadata(url, metadataJson);
+
+  // attempt to find the publisher value in our internal db, falling back to
+  // a normalized domain name if not
+  urlMetadata.publisher = await derivePublisher(context.db, urlMetadata.url);
```

Comment:
`publisher: String` is not required currently. Would it be cleaner just to return `null` or `undefined` when it's not available, instead of populating it with the domain? Perhaps worth quickly checking if this would cause a problem in any of the callers?

Reply:
Yes, I think it would be better to separate domain and publisher for this query; it allows the caller to decide how to handle null values. For example, in the admin UI, it's more obvious for editors that a publisher name is missing if the text field is empty than if it has a subdomain.